src/loader.py

```diff
@@ -100,26 +99,49 @@ async def summarize_document(doc, client):
 }
 ```
 """.strip()
+    
+    content = {"content": doc.text, **doc.metadata}
 
-    max_retries = 5
-    attempt = 0
-    while attempt < max_retries:
-        try:
-            chat_completion = client.chat.completions.create(
-                messages=[
-                    {"role": "system", "content": PROMPT},
-                    {"role": "user", "content": json.dumps(doc)},
-                ],
-                model="llama-3.1-70b-versatile",
-                response_format={"type": "json_object"},
-                temperature=0,
-            )
-            break
-        except Exception as e:
-            print("Error status {}".format(e.status_code))
-            attempt += 1
+    client = ollama.AsyncClient()
+    chat_completion = await client.chat(
+        messages = [
+            {
+                "role": "system",
+                "content": PROMPT},
+            {
+                "role": "user",
+                "content": json.dumps(content),
+            },
+        ],
+        model = "llama3.2",
+        format = "json",
+        stream = False,
+        options = {
+            "temperature": 0
+        }
+    )
 
-    summary = json.loads(chat_completion.choices[0].message.content)
+
+    # while attempt < max_retries:
+    #     try:
+    #         chat_completion = client.chat.completions.create(
+    #             messages=[
+    #                 {"role": "system", "content": PROMPT},
+    #                 {"role": "user", "content": json.dumps(doc)},
+    #             ],
+    #             model="llama-3.1-70b-versatile",
+    #             response_format={"type": "json_object"},
+    #             temperature=0,
+    #         )
+    #         break
+    #     except Exception as e:
+    #         print("Error status {}".format(e.status_code))
+    #         attempt += 1
+
+    # TODO: summary = json.loads(chat_completion.message.content)
```

Comment:
Any reason to include this large commented block?

Reply:
I forgot to remove it - I kept it there for my own reference.

Regarding this pull request.,I created it by accident but then considered it might be useful seeing so many people mentioning incognito mode. I took an interest in your project but I am far from a confident developer :')